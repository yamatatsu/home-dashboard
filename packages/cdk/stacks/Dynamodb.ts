import * as cdk from "@aws-cdk/core";
import * as dynamodb from "@aws-cdk/aws-dynamodb";

export class DynamodbStack extends cdk.Stack {
  public readonly homeDataTable: dynamodb.ITable;

  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const homeDataTable = new dynamodb.Table(this, "HomeDataTable", {
      partitionKey: {
        name: "partitionKey",
        type: dynamodb.AttributeType.STRING,
      },
      sortKey: { name: "sortKey", type: dynamodb.AttributeType.STRING },
      timeToLiveAttribute: "ttl",
      removalPolicy: cdk.RemovalPolicy.DESTROY,
    });

    this.homeDataTable = homeDataTable;
  }
}
