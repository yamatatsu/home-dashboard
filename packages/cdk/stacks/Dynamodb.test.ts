import { SynthUtils } from "@aws-cdk/assert";
import * as cdk from "@aws-cdk/core";
import { DynamodbStack } from "./Dynamodb";

test("DynamodbStack Stack", () => {
  const app = new cdk.App();

  const stack = new DynamodbStack(app, "MyTestStack", { dev: false });

  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
