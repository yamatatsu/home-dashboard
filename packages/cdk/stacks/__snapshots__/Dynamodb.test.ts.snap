// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamodbStack Stack 1`] = `
Object {
  "Resources": Object {
    "HomeDataTable592412A1": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "partitionKey",
            "AttributeType": "S",
          },
          Object {
            "AttributeName": "sortKey",
            "AttributeType": "S",
          },
        ],
        "KeySchema": Array [
          Object {
            "AttributeName": "partitionKey",
            "KeyType": "HASH",
          },
          Object {
            "AttributeName": "sortKey",
            "KeyType": "RANGE",
          },
        ],
        "ProvisionedThroughput": Object {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
        "TimeToLiveSpecification": Object {
          "AttributeName": "ttl",
          "Enabled": true,
        },
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
